corgi/tasks/prod_defs.py

```diff
@@ -63,192 +63,203 @@ def update_products() -> None:
-                )
-                product_version, _ = ProductVersion.objects.update_or_create(
-                    name=name,
-                    defaults={
-                        "version": version,
-                        "description": description,
-                        "products": product,
-                        "meta_attr": pd_product_version,
-                    },
-                )
-                product_version_node, _ = ProductNode.objects.get_or_create(
-                    object_id=product_version.pk,
-                    defaults={
-                        "parent": product_node,
-                        "obj": product_version,
-                    },
-                )
+def parse_product_version(pd_product_version, product, product_node):
+    pd_product_streams = pd_product_version.pop("product_streams", [])
+    name = pd_product_version.pop("id")
+    if match_version := RE_VERSION_LIKE_STRING.search(name):
+        version = match_version.group()
+    description = pd_product_version.pop("public_description", [])
+    logger.debug(
+        "Creating or updating Product Version: name=%s, description=%s",
+        name,
+        description,
+    )
+    product_version, _ = ProductVersion.objects.update_or_create(
+        name=name,
+        defaults={
+            "version": version,
+            "description": description,
+            "products": product,
+            "meta_attr": pd_product_version,
+        },
+    )
+    product_version_node, _ = ProductNode.objects.get_or_create(
+        object_id=product_version.pk,
+        defaults={
+            "parent": product_node,
+            "obj": product_version,
+        },
+    )
+    for pd_product_stream in pd_product_streams:
+        version = parse_stream(pd_product_stream, product, product_version, product_version_node, version)
+    product_version.save_product_taxonomy()
 
-                for pd_product_stream in pd_product_streams:
-                    active = pd_product_stream.pop("active")
-                    errata_info = pd_product_stream.pop("errata_info", [])
-                    brew_tags = pd_product_stream.pop("brew_tags", [])
-                    yum_repos = pd_product_stream.pop("yum_repositories", [])
-                    composes = pd_product_stream.pop("composes", [])
 
-                    brew_tags_dict = {}
-                    for brew_tag in brew_tags:
-                        brew_tags_dict[brew_tag["tag"]] = brew_tag["inherit"]
+def parse_stream(pd_product_stream, product, product_version, product_version_node, version):
+    active = pd_product_stream.pop("active")
+    errata_info = pd_product_stream.pop("errata_info", [])
+    brew_tags = pd_product_stream.pop("brew_tags", [])
+    yum_repos = pd_product_stream.pop("yum_repositories", [])
+    composes = pd_product_stream.pop("composes", [])
+    brew_tags_dict = {}
+    for brew_tag in brew_tags:
+        brew_tags_dict[brew_tag["tag"]] = brew_tag["inherit"]
```

Comment:
Minor nit, let's just build the dict directly when it's simple to do so / we already know what it should contain. This also helps mypy automatically infer what the type should be without needing to manually add an annotation:
```suggestion
    brew_tags_dict = {
        brew_tag["tag"]: brew_tag["inherit"]
        for brew_tag in brew_tags
    }
```

Reply:
ok